nexus/library/quantum_package_input.py

```diff
@@ -0,0 +1,786 @@
+##################################################################
+##  (c) Copyright 2018-  by Jaron T. Krogel                     ##
+##################################################################
+
+
+#====================================================================#
+#  quantum_package_input.py                                          #
+#    Supports I/O for Quantum Package input data.                    #
+#                                                                    #
+#  Content summary:                                                  #
+#    QuantumPackageInput                                             #
+#      SimulationInput class for Quantum Package.                    #
+#                                                                    #
+#    generate_quantum_package_input                                  #
+#      User-facing function to create arbitrary input.               #
+#====================================================================#
+
+import os
+from generic import obj
+from developer import DevBase,log,error
+from structure import Structure
+from physical_system import PhysicalSystem
+from simulation import SimulationInput
+from execute import execute
+
+
+bool_values = dict(T=True,F=False)
+bool_values_inv = {True:'T',False:'F'}
+
+def read_qp_value_type(value_filepath):
+    f = open(value_filepath,'r')
+    svalue = f.read().strip()
+    f.close()
+    if svalue in bool_values:
+        return 'bool'
+    else:
+        try:
+            v = int(svalue)
+            return 'int'
+        except:
+            try:
+                v = float(svalue)
+                return 'float'
+            except:
+                return 'str'
+            #end try
+        #end try
+    #end try
+#end def read_qp_value_type
+
+
+def read_qp_value(value_filepath):
+    f = open(value_filepath,'r')
+    svalue = f.read().strip()
+    f.close()
+    if svalue in bool_values:
+        v = bool_values[svalue]
+    else:
+        try:
+            v = int(svalue)
+        except:
+            try:
+                v = float(svalue)
+            except:
+                v = svalue
+            #end try
+        #end try
+    #end try
+    return v
+#end def read_qp_value
+
+
+def write_qp_value(value_filepath,value):
+    if isinstance(value,bool):
+        svalue = bool_values_inv[value]
+    elif isinstance(value,int):
+        svalue = str(value)
+    elif isinstance(value,float):
+        '{0: 24.15e}'.format(value)
+    elif isinstance(value,str):
+        svalue = value
+    else:
+        QuantumPackageInput.class_error('invalid type encountered on write\nattempted to write variable: {0}\nwith type: {1}\nvalid type options: bool,int,float,str'.format(value_filepath,value.__class__.__name__))
+    #end if
+    f = open(value_filepath,'w')
+    f.write(svalue+'\n')
+    f.close()
+#end def write_qp_value
+
+
+
```

Comment:
This looks like it comes from extract_input_specification?  If so, there should a comment about where this piece of code came from.

Reply:
I mean the code following this comment.  The email message sent a bit of code from before this comment.

Reply:
Comment added.